coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
I used a normal JS promise, and it still works. Something in my specific Ember app, makes it break, as the binding is messed up. Could we pretty please just merge it in with the fat arrow, as it makes it work for me? It should work for everyone. I think since it wraps it in another function, my bind doesn't get applied to it, and it still keeps the ```this``` reference.